hs_core/models.py

```diff
@@ -1677,6 +1936,18 @@ def get_hs_term_dict(self):
 
         return hs_term_dict
 
+    def check_folder_creation(self, folder_full_path):
+        """this checks if it is allowed to create a folder by the given path
+        if not allowed then raises ValidationError. Subclasses need to override this method
+        """
+        pass
+
+    def check_move_or_rename_file_or_folder(self, src_full_path, tgt_full_path):
```

Comment:
Same comment. It's a generic function that applies to all operations on the resource. The code to support that doesn't exist yet, though.